src/openrct2/park/ParkFile.cpp

```diff
@@ -2630,14 +2630,26 @@ namespace OpenRCT2
     template<typename T>
     void ParkFile::WriteEntitiesOfType(OrcaStream& os, OrcaStream::ChunkStream& cs)
     {
-        uint16_t count = GetEntityListCount(T::cEntityType);
+        const auto entityList = EntityList<T>();
+        // We do not rely on GetEntityListCount as there is a potential issue (#23636) that needs investigating.
+        const auto count = [&]() -> uint16_t {
+            uint16_t res = 0;
+            for ([[maybe_unused]] auto* _ : entityList)
+            {
+                res++;
+            }
+            return res;
+        }();
         cs.Write(T::cEntityType);
         cs.Write(count);
         for (auto* ent : EntityList<T>())
         {
             cs.Write(ent->Id);
             ReadWriteEntity(os, cs, *ent);
         }
+        Guard::Assert(
+            count == GetEntityListCount(T::cEntityType),
+            "Please submit a new issue on GitHub with the latest working auto save file");
```

Comment:
I don’t think we should crash during saving and lose people’s data. That’s about the worst thing we can do.

Instead, it should just be a warning. And automatically create a report on Backtrace, if that’s possible to do without crashing.

Reply:
This is already corrupting their saves so I don't see the the issue. The point is to give them us a most recent working auto save in case this event happens. It is also not even clear if this is the actual issue or not, but I can see how a hard crash is sub optimal.

Reply:
This is why it asks for the last auto save as the state its trying to save is useless in an already corrupted state.